src/microns_to_nwb/tools/ophys/ophys.py

```diff
@@ -0,0 +1,113 @@
+
+    correlation, average = (nda.SummaryImages & field_key).fetch1("correlation", "average")
+
+    correlation_image = GrayscaleImage(correlation)
+    average_image = GrayscaleImage(average)
+
+    ophys.add(correlation_image)
+    ophys.add(average_image)
+
+
+def add_plane_segmentation(field_key, nwb, imaging_plane, image_segmentation):
+    ps = image_segmentation.create_plane_segmentation(
+        name=f"PlaneSegmentation{field_key['field']}",
+        description="output from segmenting my favorite imaging plane",
+        imaging_plane=imaging_plane,
+    )
+    ps.add_column("mask_type", "type of ROI")
+
+    image_height, image_width = (nda.Field & field_key).fetch1("px_height", "px_width")
+
+    mask_pixels, mask_weights, mask_ids, mask_types = (nda.Segmentation * nda.MaskClassification & field_key).fetch(
+        "pixels", "weights", "mask_id", "mask_type", order_by="mask_id"
+    )
+
+    # Reshape masks
+    masks = func.reshape_masks(mask_pixels, mask_weights, image_height, image_width)
+
+    for image_mask, mask_id, mask_type in zip(masks, mask_ids, mask_types):
+        ps.add_roi(
+            image_mask=image_mask,
+            id=mask_id,
+            mask_type=mask_type,
+        )
+
+    return ps
+
+
+def add_roi_response_series(scan_key, field_key, nwb, plane_segmentation):
+    frame_times = (nda.FrameTimes & scan_key).fetch1("frame_times")
+
+    data = np.vstack((nda.Fluorescence() & field_key).fetch("trace", order_by="mask_id")).T
+
+    rt_region = plane_segmentation.create_roi_table_region(
+        region=list(range(data.shape[1])), description=f"all rois in field {field_key['field']}"
+    )
+
+    roi_response_series = RoiResponseSeries(
+        name=f"RioResponseSeries{field_key['field']}",
+        description=f"traces for field {field_key['field']}",
+        data=H5DataIO(data, compression=True),
+        rois=rt_region,
+        timestamps=H5DataIO(frame_times, compression=True),
+        units="n/a",
+    )
+
+    fluorescence = Fluorescence()
+    fluorescence.add(roi_response_series)
+
+
+def add_ophys(scan_key, nwb):
+    device = nwb.create_device(
+        name="Microscope",
+        description="two-photon random access mesoscope",
+    )
+    ophys = check_module(nwb, "ophys")
+    image_segmentation = ImageSegmentation()
+    ophys.add(image_segmentation)
+    all_field_data = (nda.Field & scan_key).fetch(as_dict=True)
+    for field_data in all_field_data:
+        optical_channel = OpticalChannel(
+            name="OpticalChannel",
+            description="an optical channel",
+            emission_lambda=500.0,
+        )
+        imaging_plane = nwb.create_imaging_plane(
+            name=f"ImagingPlane{field_data['field']}",
+            optical_channel=optical_channel,
+            imaging_rate=np.nan,
+            description="no description",
+            device=device,
+            excitation_lambda=920.0,
+            indicator="GCaMP6",
+            location="unknown",
```

Comment:
From section 2.24 this ought to be know, right? Are those fields available in the database?

Reply:
From [the article](https://www.biorxiv.org/content/10.1101/2021.07.28.454025v2.full.pdf)
> The center of the volume was placed at the junction of primary visual cortex (VISp) and three higher visual areas, lateromedial area (VISlm), rostrolateral area (VISrl) and anterolateral area (VISal) in order to image retinotopically-matched neurons connected via long-range inter-areal feedforward and feedback connections.

Reply:
location = {'AL', 'LM', 'RL', 'V1'}

Reply:
Last thing to check is if those terms are proper in an Allen Institute ontology for brain areas (I'm assuming they are, but want to triple-check)

Reply:
Yes, perfect! Thank you for checking